var/spack/repos/builtin/packages/mapserver/package.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+
+
+class Mapserver(CMakePackage):
+    """MapServer is an Open Source geographic data rendering engine written
+       in C. Beyond browsing GIS data, MapServer allows you create
+       "geographic image maps", that is, maps that can direct users
+       to content"""
+
+    homepage = "http://www.mapserver.org/"
+    url      = "http://download.osgeo.org/mapserver/mapserver-7.2.1.tar.gz"
+
+    version('7.2.1', sha256='9459a7057d5a85be66a41096a5d804f74665381186c37077c94b56e784db6102')
+
+    depends_on('libpng')
+    depends_on('freetype')
+    depends_on('jpeg')
+    depends_on('zlib')
+    depends_on('proj')
+    depends_on('curl', when='+curl')
+    depends_on('geos')
+    depends_on('libxml2')
+    depends_on('gdal')
+    depends_on('swig', type='build')
+    depends_on('python', when='+python')
+    depends_on('ruby', when='+ruby')
+    depends_on('java', when='+java')
+    depends_on('perl', when='+perl')
+
+    variant('python', default=False)
+    variant('curl', default=False)
+    variant('ruby', default=False)
+    variant('java', default=False)
+    variant('perl', default=False)
+
+    extends('python', when='+python')
+
+    @when('+python')
+    def patch(self):
+        # The Python bindings install themselves into the main python
+        # site-packages directory, instead of under the current package
+        # prefix. This hack patches the CMakeLists.txt for the Python
+        # bindings and hard-wires in the right destination. A bit ugly,
+        # sorry, but I don't speak cmake.
+        pyversiondir = "python{0}".format(self.spec['python'].version.up_to(2))
+        sitepackages = os.path.join(self.spec.prefix.lib,
+                                    pyversiondir,
+                                    "site-packages")
+        filter_file('\${PYTHON_SITE_PACKAGES}',
+                    sitepackages,
+                    'mapscript/python/CMakeLists.txt')
+
+    def cmake_args(self):
+        args = []
+
+        if '+python' in self.spec:
+            args.append('-DWITH_PYTHON=ON')
+        else:
+            args.append('-DWITH_PYTHON=OFF')
+
+        if '+java' in self.spec:
+            args.append('-DWITH_JAVA=ON')
+        else:
+            args.append('-DWITH_JAVA=OFF')
+
+        if '+ruby' in self.spec:
+            args.append('-DWITH_RUBY=ON')
+        else:
+            args.append('-DWITH_RUBY=OFF')
+
+        if '+perl' in self.spec:
+            args.append('-DWITH_PERL=ON')
+        else:
+            args.append('-DWITH_PERL=OFF')
+
+        # These things are switched on by default, although possibly some
+        # should be variants.
+        args.append('-DWITH_CURL=ON')
```

Comment:
You do have a variant for this one.

Reply:
Oops :-) Thank you for catching my omissions